### ISSUE TITLE
ERROR: No bundle URL present.

### ISSUE THREAD

**Issue by neko3cs** [microsoft/react-native-macos#2108]
### Environment

```markdown
react-native -v:12.3.6(npx)
npm ls react-native-macos:react-native-macos@0.73.24
node -v:v21.7.2
npm -v:10.5.0
yarn --version:1.22.22
xcodebuild -version:Xcode 15.3 Build version 15E204a
```

### Steps to reproduce the bug

Do this getting started.
https://microsoft.github.io/react-native-windows/docs/rnm-getting-started

Before `npx react-native run-macos`, you should run `cd macos && pod install`.
I cannot run app without this step.

### Expected Behavior

Launch application and show ui same as iOS.

### Actual Behavior

<img width="1278" alt="image" src="https://github.com/microsoft/react-native-macos/assets/36363130/968068ea-3c9f-4b56-9776-b164347499e3">

### Reproducible Demo

[src.zip](https://github.com/microsoft/react-native-macos/files/14896219/src.zip)

Full size is over 25MB.
So this is a sample deleted this:
- macos/Pod
  - you can get with: `cd macos && pod install`
- node_module
  - you can get with: `yarn install`

### Additional context

I tryed what this comment telling, but I cannot launch app.
https://github.com/microsoft/react-native-macos/issues/400#issuecomment-632348718

First, `--use-react-native-macos` has occured not found error.
error messeage: `error: unknown option '--use-react-native-macos'`

So I  removed 👆 option and  run this: `node node_modules/react-native-macos/local-cli/cli.js bundle --entry-file index.js --platform macos --bundle-output MyApp.jsbundle`
But same error has occured.

**Comment by Saadnajmi** [comment]
Yeah that option is deprecated / removed so we should update the docs. 
Does your app work if you run a metro server locally (I.e run 'npx react-native start' from a terminal while the app is running)?

**Comment by neko3cs** [comment]
@Saadnajmi 
I'm sorry for late message.
iOS App is work. but macOS App is not work.
To begin with, when I run `npx react-native start`, I don't see the macOS startup options.
Only "reload the app", "open developer menu", "run on iOS", "run on Android".

**Comment by Saadnajmi** [comment]
@neko3cs The macOS startup options won't show, but if you click "Reload" in the Redbox, it should still properly reload the bundle

**Comment by neko3cs** [comment]
@Saadnajmi 
I think We have difference in perception. The image of section "Actual Behavior" is what running "npx react-native run-macos", not "npx react-native start". Both of that commands behave differently.

![react-native-macos-error](https://github.com/microsoft/react-native-macos/assets/36363130/cb73d9a4-ea16-48ac-bd50-33f22a82d13d)

**Comment by bear-ei** [comment]
> npx react-native start

Currently, executing npx react-native run-macos in version 0.73.0 does indeed result in the issue of "No bundle URL present." However, you can first run npx react-native start in one terminal to manually start the local Metro server, and then execute npx react-native run-macos to start the macOS application.

**Comment by bear-ei** [comment]
> @Saadnajmi I think We have difference in perception. The image of section "Actual Behavior" is what running "npx react-native run-macos", not "npx react-native start". Both of that commands behave differently.


Currently, executing npx react-native run-macos in version 0.73.0 does indeed result in the issue of "No bundle URL present." However, you can first run npx react-native start in one terminal to manually start the local Metro server, and then execute npx react-native run-macos to start the macOS application.

**Comment by Saadnajmi** [comment]
@bear-ai Thanks, yes I think `run-macos` is broken atm, I want to replace it with https://github.com/react-native-community/cli/tree/main/packages/cli-platform-apple , perhaps with the 0.74 release. I'm glad we found a workaround!

**Comment by neko3cs** [comment]
Yeeeeeaaaaaah!!! I'm happy to lauch my app!
@Saadnajmi @bear-ei Thank you so much for your helping!
I ran metro server on background(`npx react-native start` on other terminal), and I ran `npx react-native run-macos`.
So app is launched.

I hope to fix this problem. Is this a Metro server issue?
Should I open an Issue in [the Repository](https://github.com/react-native-community/cli/tree/main/packages/cli-platform-apple)?

**Comment by michelcrypt4d4mus** [comment]
also having this issue

**Comment by dannyxu2015** [comment]
solution of @bear-ei worked in 0.71.0, which I new project by referring the [official document](https://microsoft.github.io/react-native-windows/docs/rnm-getting-started).

But if I new a project with latest version(0.76.5) by command: `npx react-native@latest init test2`, following error occured:
```
✔ Downloading template
✔ Copying template
✔ Processing template
✔ Installing dependencies
✔ Do you want to install CocoaPods now? Only needed if you run your project in Xcode directly … yes
✔ Installing Ruby Gems
✖ Installing CocoaPods dependencies with New Architecture (this may take a few minutes)
error   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 23.3M  100 23.3M    0     0   368k      0  0:01:04  0:01:04 --:--:--  371k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16.3M  100 16.3M    0     0   365k      0  0:00:45  0:00:45 --:--:--  369k

[!] [Codegen] warn: using experimental new codegen integration
error Installing Cocoapods failed. This doesn't affect project initialization and you can safely proceed. However, you will need to install Cocoapods manually when running iOS, follow additional steps in "Run instructions for iOS" section.

Error: Looks like your iOS environment is not properly set. Please go to https://reactnative.dev/docs/environment-setup?os=macos&platform=ios&guide=native and follow the React Native CLI QuickStart guide for macOS and iOS.
```

**Comment by bear-ei** [comment]
> Installing Cocoapods failed. This doesn't affect project initialization and you can safely proceed. However, you will need to install Cocoapods manually when running iOS, follow additional steps in "Run instructions for iOS" section

I think that for questions related to `react-native-cli`, you should refer to the main branch of [react native](https://github.com/facebook/react-native/issues). Because the error message clearly mentions that you are using `react-native-cli` to initialize a `react native` project. At this stage, it has nothing to do with `react-native-macos` currently. `"Installing Cocoapods failed"` usually occurs due to problems with your network (such as using a VPN in a specific region) or the configuration of your local development environment.

**Comment by dannyxu2015** [comment]
yeah, fixed it by change directory into the 'ios' , run command `bundle install` and `bundle exec pod install`, and then change to app project directory, init react native macos. Thanks @bear-ei
